libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -448,6 +449,33 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
                           rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
             // TODO implement joystick RC control to AP
+            const uint32_t tnow = AP_HAL::millis();
+            
+            for (uint8_t i=0; i<rx_joy_topic.axes_size; i++) {
+                const uint16_t override_data = static_cast<uint16_t>(rx_joy_topic.axes[i]);
```

Comment:
The override_data should be cast to `float` as we are expecting it to be in `[-1.0f, 1.0f].

Maybe rename variable to `const float axes_value`.

Reply:
We need to compare the `override_data` with `UINT16_MAX` as per mavlink specs - ` if (override_data != UINT16_MAX) {` so I think keeping the cast to `uint16_t` would be better.